### PR TITLE
MAUI-3110 Prepared UG for legend and tooltip customization support

### PR DESCRIPTION
**Pull request by TamilarasanGunasekaran**
Hi @vigneshrameshsync ,
I have prepared UG content for legend and tooltip customization support.

### CODE REVIEW

File: MAUI/Maps/tooltip.md [hunk 1]
```diff
@@ -271,6 +271,110 @@ The [`ShowMarkerTooltip`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Ma
 
 ## Appearance customization
 
+You can customize the below appearances of the tooltip.
+
+* **Background** - Change the background color of the tooltip in the maps using the MapTooltipSettings.Background property.
+* **Padding** - Change the padding of tooltip text in the maps using the MaptooltipSettings.Padding property.
+* **TextStyle** - Change the tooltip text appearance in the maps using the MapTooltipSettings.TextStyle property
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tooltip.md:278]: Change the tooltip text appearance such as `TextColor`, `FontSize`, and `FontAttributes` in the maps using `MapTooltipSettings.TextStyle` property.

File: MAUI/Maps/legend.md [hunk 2]
```diff
@@ -660,4 +660,169 @@ public class Model
 
 ![Legend position](images/legend/legend-position.png)
 
+## Legend icon customization
+
+You can customize the legend icon using the `IconType` and `IconSize` properties. The default value of `IconType` is [`ShapeType.Rectangle`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Core.ShapeType.html#Syncfusion_Maui_Core_ShapeType_Rectangle) and 12 is the default value of `IconSize`.
```
Comment (Chrispineagunja) [inline MAUI/Maps/legend.md:665]: Customize the legend icon using the `IconType` and `IconSize` properties. The default value of the `IconType` is [`ShapeType.Rectangle`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Core.ShapeType.html#Syncfusion_Maui_Core_ShapeType_Rectangle) and 12 is the default value of the `IconSize.`

File: MAUI/Maps/tooltip.md [hunk 3]
```diff
@@ -383,7 +487,7 @@ Data template can be used customize the tooltip view using [`ShapeTooltipTemplat
             };
             Binding binding = new Binding();
             binding.Source = grid.BindingContext;
-            binding.Path = nameof(Model1.Continent);
+            binding.Path = nameof(DataItem.Continent);
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tooltip.md:490]: The syntax is wrong. Please refer

https://github.com/essential-studio/maui-maps/blob/release/20.3.0.1/maui/src/maps/ShapeLayer/MapShapeLayer.cs#L211

File: MAUI/Maps/legend.md [hunk 4]
```diff
@@ -660,4 +660,169 @@ public class Model
...
+    }
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Legend icon customization](images/legend/legend-icon-customization.png)
+
 N> You can refer to our [.NET MAUI Maps](https://www.syncfusion.com/maui-controls/maui-maps) feature tour page for its groundbreaking feature representations. You can also explore our [.NET MAUI Maps Legend example](https://github.com/syncfusion/maui-demos/) that shows how to configure a Maps in .NET MAUI.
```
Comment (Chrispineagunja) [inline MAUI/Maps/legend.md:828]: N> Refer to [.NET MAUI Maps](https://www.syncfusion.com/maui-controls/maui-maps) feature tour page for its groundbreaking feature representations. Also, explore [.NET MAUI Maps Legend example](https://github.com/syncfusion/maui-demos/), which shows how to configure Maps in .NET MAUI.

File: MAUI/Maps/tooltip.md [hunk 5]
```diff
@@ -445,8 +549,13 @@ Data template can be used customize the tooltip view using [`ShapeTooltipTemplat
 ![Maps tooltip appearance customization](images/tooltip/tooltip_textStyle.png)
 
 N>
-* The BindingContext of the Shape and Bubble will be the corresponding underline object.
-* The BindingContext of the Marker will be the corresponding MapMarker.
+* The BindingContext of the Tooltip will be the `TooltipInfo`, it holds the `DataItem` property. This property has corresponding underline object for shape, bubble and marker.
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tooltip.md:552]: TooltipInfo -> MapTooltipInfo.

File: MAUI/Maps/tooltip.md [hunk 6]
```diff
@@ -271,6 +271,110 @@ The [`ShowMarkerTooltip`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Ma
 
 ## Appearance customization
 
+You can customize the below appearances of the tooltip.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:274]: Customize the following appearances of the tooltip.

File: MAUI/Maps/tooltip.md [hunk 7]
```diff
@@ -271,6 +271,110 @@ The [`ShowMarkerTooltip`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Ma
...
+
+* **Background** - Change the background color of the tooltip in the maps using the MapTooltipSettings.Background property.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:276]: * **Background** - Change the background color of the tooltip in maps using the MapTooltipSettings.Background property.

File: MAUI/Maps/tooltip.md [hunk 8]
```diff
@@ -271,6 +271,110 @@ The [`ShowMarkerTooltip`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Ma
...
+* **Padding** - Change the padding of tooltip text in the maps using the MaptooltipSettings.Padding property.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:277]: * **Padding** - Change the padding of the tooltip text in maps using the MaptooltipSettings.Padding property.

File: MAUI/Maps/tooltip.md (hunk 1 above)
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:278]: * **TextStyle** - Change the tooltip text appearance in maps using the MapTooltipSettings.TextStyle property.

File: MAUI/Maps/tooltip.md [hunk 9]
```diff
@@ -271,6 +271,110 @@ The [`ShowMarkerTooltip`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Ma
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Maps tooltip settings](images/tooltip/tooltip_appearance.png)
+
 ### Apply data template for tooltip
 
 Data template can be used customize the tooltip view using [`ShapeTooltipTemplate`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Maps.MapShapeLayer.html#Syncfusion_Maui_Maps_MapShapeLayer_ShapeTooltipTemplate) property. The following code example shows the usage of `DataTemplate`.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:380]: Data template can be used to customize the tooltip view using the [`ShapeTooltipTemplate`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Maps.MapShapeLayer.html#Syncfusion_Maui_Maps_MapShapeLayer_ShapeTooltipTemplate) property. The following code example shows the usage of the `DataTemplate.`

File: MAUI/Maps/tooltip.md (hunk 5 above)
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:552]: * The BindingContext of the Tooltip will be the `TooltipInfo,` which holds the `DataItem` property. This property has a corresponding underline object for shape, bubble, and marker.

File: MAUI/Maps/tooltip.md [hunk 10]
```diff
@@ -445,8 +549,13 @@ Data template can be used customize the tooltip view using [`ShapeTooltipTemplat
 ![Maps tooltip appearance customization](images/tooltip/tooltip_textStyle.png)
 
 N>
-* The BindingContext of the Shape and Bubble will be the corresponding underline object.
-* The BindingContext of the Marker will be the corresponding MapMarker.
+* The BindingContext of the Tooltip will be the `TooltipInfo`, it holds the `DataItem` property. This property has corresponding underline object for shape, bubble and marker.
+
+N>
+* Refer to the `ShapeTooltipSettings` for customize the shape tooltip appearance.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:555]: * Refer to the `ShapeTooltipSettings` to customize the shape tooltip appearance.

File: MAUI/Maps/tooltip.md [hunk 11]
```diff
@@ -445,8 +549,13 @@ Data template can be used customize the tooltip view using [`ShapeTooltipTemplat
...
+* Refer to the `BubbleTooltipSettings` for customize the bubble tooltip appearance.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:556]: * Refer to the `BubbleTooltipSettings` to customize the bubble tooltip appearance.

File: MAUI/Maps/tooltip.md [hunk 12]
```diff
@@ -445,8 +549,13 @@ Data template can be used customize the tooltip view using [`ShapeTooltipTemplat
...
+* Refer to the `MarkerTooltipSettings` for customize the marker tooltip appearance.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:557]: * Refer to the `MarkerTooltipSettings` to customize the marker tooltip appearance.

File: MAUI/Maps/legend.md (hunk 2 above)
Comment (vigneshrameshsync) [inline MAUI/Maps/legend.md:665]: You can customize the legend icon using the `IconType` and `IconSize` properties. The default value of `IconType` and  `IconSize` are [`ShapeType.Rectangle`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Core.ShapeType.html#Syncfusion_Maui_Core_ShapeType_Rectangle) and 12 respectively.

File: MAUI/Maps/tooltip.md [hunk 13]
```diff
@@ -445,8 +549,13 @@ Data template can be used customize the tooltip view using [`ShapeTooltipTemplat
...
+
 
 N>
 * Refer to the [`ShapeTooltipTemplate`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Maps.MapShapeLayer.html#Syncfusion_Maui_Maps_MapShapeLayer_ShapeTooltipTemplate), for customize tooltip for the shapes.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:561]: * Refer to the [`ShapeTooltipTemplate`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Maps.MapShapeLayer.html#Syncfusion_Maui_Maps_MapShapeLayer_ShapeTooltipTemplate) to customize the tooltip for the shapes.

File: MAUI/Maps/legend.md [hunk 14]
```diff
@@ -660,4 +660,169 @@ public class Model
...
+        State = state;
+        Population = population;
+    }
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Legend icon customization](images/legend/legend-icon-customization.png)
```
Comment (vigneshrameshsync) [inline MAUI/Maps/legend.md:826]: Update the image as discussed.

File: MAUI/Maps/tooltip.md [hunk 15]
```diff
@@ -271,6 +271,110 @@ The [`ShowMarkerTooltip`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Ma
...
+        State = state;
+        Population = population;
+    }
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Maps tooltip settings](images/tooltip/tooltip_appearance.png)
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tooltip.md:376]: Update the image as discussed.

File: MAUI/Maps/tooltip.md (hunk 1 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/tooltip.md]: I have changed.

File: MAUI/Maps/tooltip.md (hunk 3 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/tooltip.md]: I have updated.

File: MAUI/Maps/tooltip.md (hunk 5 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/tooltip.md]: I have changed.

File: MAUI/Maps/legend.md (hunk 2 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/legend.md]: I have changed.

File: MAUI/Maps/legend.md (hunk 14 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/legend.md:829]: I have changed.

File: MAUI/Maps/tooltip.md (hunk 15 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/tooltip.md]: I have updated.

File: MAUI/Maps/legend.md [hunk 16]
```diff
@@ -660,4 +660,172 @@ public class Model
 
 ![Legend position](images/legend/legend-position.png)
 
+## Appearance customization
+
+You can customize the legend items using the following properties.
```
Comment (Chrispineagunja) [inline MAUI/Maps/legend.md:665]: Customize the legend items using the following properties.

File: MAUI/Maps/tooltip.md [hunk 17]
```diff
@@ -271,6 +271,110 @@ The [`ShowMarkerTooltip`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Ma
 
 ## Appearance customization
 
+You can customize the below appearances of the tooltip.
+
+* **Background** - Change the background color of the tooltip in the maps using the `MapTooltipSettings.Background` property.
+* **Padding** - Change the padding of tooltip text in the maps using the `MaptooltipSettings.Padding` property.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:277]: * **Padding** - Change the padding of the tooltip text in the maps using the `MaptooltipSettings.Padding` property.

File: MAUI/Maps/tooltip.md [hunk 18]
```diff
@@ -271,6 +271,110 @@ The [`ShowMarkerTooltip`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Ma
...
+* **TextStyle** - Change the tooltip text appearance such as `TextColor`, `FontSize`, `FontAttributes` and `FontFamily` in the maps using `MapTooltipSettings.TextStyle` property.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:278]: * **TextStyle** - Change the tooltip text appearance such as `TextColor,` `FontSize,` `FontAttributes,` and `FontFamily` in the maps using the `MapTooltipSettings.TextStyle` property.

File: MAUI/Maps/tooltip.md [hunk 19]
```diff
@@ -445,8 +555,13 @@ Data template can be used customize the tooltip view using [`ShapeTooltipTemplat
 ![Maps tooltip appearance customization](images/tooltip/tooltip_textStyle.png)
 
 N>
-* The BindingContext of the Shape and Bubble will be the corresponding underline object.
-* The BindingContext of the Marker will be the corresponding MapMarker.
+* The `BindingContext` of the Tooltip will be the `MapTooltipInfo`, and it has the `DataItem` property. This property holds the corresponding underline object for shape, bubble tooltip and holds the corresponding `MapMarker` for the marker tooltip.
```
Comment (Chrispineagunja) [inline MAUI/Maps/tooltip.md:558]: * The `BindingContext` of the Tooltip will be the `MapTooltipInfo,` and it has the `DataItem` property. This property holds the corresponding underline object for the shape, bubble tooltip, and the corresponding `MapMarker` for the marker tooltip.

File: MAUI/Maps/legend.md (hunk 16 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/legend.md:665]: I have changed.

File: MAUI/Maps/tooltip.md (hunk 17 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/tooltip.md:277]: I have changed.

File: MAUI/Maps/tooltip.md (hunk 18 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/tooltip.md:278]: I have changed.

File: MAUI/Maps/tooltip.md (hunk 19 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/tooltip.md:558]: I have changed.

File: MAUI/Maps/tooltip.md (hunk 13 above)
Comment (TamilarasanGunasekaran) [reply MAUI/Maps/tooltip.md:567]: I have changed.

**Review by vigneshrameshsync — APPROVED** [review]
The corrections are fine.

**Pull request merged**